TLM/TLM/Util/Extensions/NetLaneExtensions.cs

```diff
@@ -7,23 +7,53 @@ public static class NetLaneExtensions {
         internal static ref NetLane ToLane(this uint laneId) => ref _laneBuffer[laneId];
 
         /// <summary>
-        /// Checks if the netLane is Created, but not Deleted.
+        /// Checks <paramref name="netLane"/> is <c>Created</c> but not <c>Deleted</c>.
         /// </summary>
         /// <param name="netLane">netLane</param>
-        /// <returns>True if the netLane is valid, otherwise false.</returns>
+        /// <returns>Returns <c>true</c> if valid, otherwise <c>false</c>.</returns>
         public static bool IsValid(this ref NetLane netLane) =>
             ((NetLane.Flags)netLane.m_flags).CheckFlags(
                 required: NetLane.Flags.Created,
                 forbidden: NetLane.Flags.Deleted);
 
         /// <summary>
-        /// Checks if the netLane is Created, but not Deleted and if its netSegment is Created, but neither Collapsed nor Deleted.
+        /// Checks <paramref name="laneId"/> is not <c>0</c>,
+        /// then checks netLane is <c>Created</c> but not <c>Deleted</c>.
+        /// </summary>
+        /// <param name="laneId">The id of the lane to check.</param>
+        /// <returns>Returns <c>true</c> if valid, otherwise <c>false</c>.</returns>
+        public static bool IsValid(this uint laneId)
+            => (laneId != 0) && laneId.ToLane().IsValid();
+
+        /// <summary>
+        /// Checks <paramref name="netLane"/> is <c>Created</c> but not <c>Deleted</c>,
+        /// then checks its netSegment is <c>Created</c> but not <c>Collapsed|Deleted</c>.
         /// </summary>
         /// <param name="netLane">netLane</param>
-        /// <returns>True if the netLane and its netSegment is valid, otherwise false.</returns>
+        /// <returns>Returns <c>true</c> if valid, otherwise <c>false</c>.</returns>
         public static bool IsValidWithSegment(this ref NetLane netLane) {
             return netLane.IsValid()
                 && netLane.m_segment.ToSegment().IsValid();
         }
+
+        /// <summary>
+        /// Checks <paramref name="laneId"/> is not <c>0</c>,
+        /// then checks netLane is <c>Created</c> but not <c>Deleted</c>,
+        /// then checks its segmentId is not <c>0</c>,
+        /// then checks its netSegment is <c>Created</c> but not <c>Collapsed|Deleted</c>.
+        /// </summary>
+        /// <param name="laneId">The id of the lane to check.</param>
+        /// <returns>Returns <c>true</c> if valid, otherwise <c>false</c>.</returns>
+        public static bool IsValidWithSegment(this uint laneId) {
```

Comment:
we already have this in lane extensions:
```cs
        public static bool IsValidWithSegment(this ref NetLane netLane) {
            return netLane.IsValid()
                && netLane.m_segment.ToSegment().IsValid();
        }
```
But it does not check for laneId < 0

there is the minor issue that we some times check for laneId < 0  and sometimes don't based on the mood of the programmer at the time!

Reply:
wait `uint` cannot be less than zero anyway so its fine.
I think we should use the old overload instead of introducing this new one.

Reply:
Yeah, I know unsigned numbers can't be negative (no idea why so much old code has the `<=` check on `ushort` and `uint`).

But regardless, that's not why we check `id != 0`, is it?

Reply:
as I said `id != 0` check is unnecessary the corresponding element is invalid anyway.